Microsoft.HealthVault.Fhir/Transformers/ObservationToHealthVault.cs

```diff
@@ -66,24 +70,52 @@ public static ThingBase ToHealthVault(this Observation observation)
             return baseThing;
         }
 
-        internal static T GetThingValueFromQuantity<T>(Quantity value) where T : Measurement<double>, new()
+        internal static T GetThingValueFromQuantity<T>(Quantity quantityValue) where T : Measurement<double>, new()
         {
-            // TODO: detect the units from the code (value.Unit)
-            if (value != null)
+            if (quantityValue?.Value == null)
             {
-                if (value.Value.HasValue)
+                return null;
+            }
+            
+            var unitConversion = UnitResolver.Instance.UnitConversions.FirstOrDefault(x => x.Code.Equals(quantityValue.Code, StringComparison.Ordinal));
+
+            double convertedValue;
+            if (unitConversion != null)
+            {
+                object unitEnum;
+                try
```

Comment:
Ok .. thanks for the clarification .. but do you want to check separately than passing to Enum .. I prefer to first check if the type exist, throw an exception if not, then pass to Enum ? you may want to do the same for the following statement for UnitsNet ? ..